frontend/public/module/k8s/pods.ts

```diff
@@ -101,12 +109,18 @@ const genericFormatter = volInfo => {
     if (key === 'readOnly') {
       return '';
     }
+    if (key === 'defaultMode') {
+      return '';
+    }
+    if (key === 'optional') {
+      return '';
+    }
     return volInfo[key];
   });
   if (keys.indexOf('readOnly') !== -1) {
     parts.push(volInfo.readOnly ? 'ro' : 'rw');
   }
-  return parts.join(' ') || null;
+  return parts.join(' ').trim() || null;
```

Comment:
Why is `trim` necessary?

Reply:
For resource types such as Secrets and ConfigMaps, there is another parameter returned called `defaultMode` which is a value denoting its permissions. In the case of a secret that I had in my cluster, the parameter was `defaultMode : 420`
So the `loc` returned was `420 <secret_name>` , On omitting the defaultMode, the value returned was ` <secret_name>` with a space in the beginning. This led to the path being incorrect and hence, the link wasn't working.